torch/utils/_sympy/numbers.py

```diff
@@ -0,0 +1,394 @@
+            if other is S.NegativeInfinity:
+                return S.NegativeInfinity
+            if other in (S.NegativeIntInfinity, S.NaN):
+                return S.NaN
+            return self
+        return Number.__add__(self, other)
+
+    __radd__ = __add__
+
+    @_sympifyit("other", NotImplemented)
+    def __sub__(self, other):
+        if isinstance(other, Number) and global_parameters.evaluate:
+            if other is S.Infinity:
+                return S.NegativeInfinity
+            if other in (S.IntInfinity, S.NaN):
+                return S.NaN
+            return self
+        return Number.__sub__(self, other)
+
+    @_sympifyit("other", NotImplemented)
+    def __rsub__(self, other):
+        return (-self).__add__(other)
+
+    @_sympifyit("other", NotImplemented)
+    def __mul__(self, other):
+        if isinstance(other, Number) and global_parameters.evaluate:
+            if other.is_zero or other is S.NaN:
+                return S.NaN
+            if other.is_extended_positive:
+                return self
+            return S.NegativeIntInfinity
+        return Number.__mul__(self, other)
+
+    __rmul__ = __mul__
+
+    @_sympifyit("other", NotImplemented)
+    def __truediv__(self, other):
+        if isinstance(other, Number) and global_parameters.evaluate:
+            if other in (
+                S.Infinity,
+                S.IntInfinity,
+                S.NegativeInfinity,
+                S.NegativeIntInfinity,
+                S.NaN,
+            ):
+                return S.NaN
+            if other.is_extended_nonnegative:
+                return S.Infinity  # truediv produces float
+            return S.NegativeInfinity  # truediv produces float
+        return Number.__truediv__(self, other)
+
+    def __abs__(self):
+        return S.IntInfinity
+
+    def __neg__(self):
+        return S.NegativeIntInfinity
+
+    def _eval_power(self, expt):
+        if expt.is_extended_positive:
+            return S.IntInfinity
+        if expt.is_extended_negative:
+            return S.Zero
+        if expt is S.NaN:
+            return S.NaN
+        if expt is S.ComplexInfinity:
+            return S.NaN
+        if expt.is_extended_real is False and expt.is_number:
+            from sympy.functions.elementary.complexes import re
+
+            expt_real = re(expt)
+            if expt_real.is_positive:
+                return S.ComplexInfinity
+            if expt_real.is_negative:
+                return S.Zero
+            if expt_real.is_zero:
+                return S.NaN
+
+            return self ** expt.evalf()
+
+    def _as_mpf_val(self, prec):
+        return mlib.finf
+
+    def __hash__(self):
+        return super().__hash__()
+
+    def __eq__(self, other):
+        return other is S.IntInfinity
+
+    def __ne__(self, other):
+        return other is not S.IntInfinity
+
+    def __gt__(self, other):
+        if other is S.Infinity:
+            return sympy.false  # sympy.oo > int_oo
```

Comment:
I get this. Looking online, I discovered that the maximum float in Python (presumably what sympy.oo is) is much larger than the maximum int.

Reply:
Not really. I think this is a stylistic choice, as there no good answer for this question. You probably want to have `sympy.oo > int_oo` AND `sympy.oo < int_oo`.
`sympy.oo` is implemented in a similar way as `int_oo`, it's a concept, it's completely detached from the ideas of representability of numbers in floating arithmetic.

Reply:
> You probably want to have sympy.oo > int_oo AND sympy.oo < int_oo.

Is there a typo here? How come I would want to have two symbols X and Y such that both X < Y and Y < X. That would create a contradiction, wouldn't it?

Reply:
Why would it? It'd just be the case if we asked `operator<` to be associated with a total order, but that is already not the case, given that `nan` exists, and `-0.0` also is a thing. `operator<`, as implemented in hardware, gives you a (strict) partial order.

Here, what we would be proposing is that with this `int_oo`, we would have that integers  are totally ordered (nice) and when combined with floats, they forma  preorder, where `int_oo` and `inf` are "equivalent elements" (remember that a preorder may be turned into a poset by quotienting out the strongly connected components). This sounds quite reasonable to me.